test/parallel/test-http2-compat-pushrequest.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const h2 = require('http2');
+
+function isEven(number) {
+  return number % 2 === 0;
+}
+
+function isOdd(number) {
+  return number % 2 === 1;
+}
+
+// Push a request & response
+
+const server = h2.createServer();
+server.listen(0, common.mustCall(function() {
+  server.once('request', common.mustCall(function(request, response) {
+    assert.ok(isOdd(response.stream.id));
+
+    response.write('This is a client-initiated response');
+    response.stream.on('finish', common.mustCall(function() {
+      server.close();
+    }));
+
+    const headers = {
+      ':path': '/pushed',
+      ':method': 'GET',
+      ':scheme': 'http',
+      ':authority': 'localhost'
+    };
+
+    const pushRequest = response.createPushRequest(headers);
+
+    pushRequest.push(common.mustCall(function(error, pushResponse) {
+      assert.strictEqual(error, null);
+      assert.ok(isEven(pushResponse.stream.id));
+
+      pushResponse.write('This is a server-initiated response');
+
+      // TODO(sebdeckers) Remove this forced delay workaround.
+      // pushResponse.end();
+      // response.end();
+      setTimeout(() => pushResponse.end(), 100);
+      setTimeout(() => response.end(), 200);
+    }));
+  }));
+
+  const url = `http://localhost:${server.address().port}`;
+  const client = h2.connect(url, common.mustCall(function() {
+    const headers = {
+      ':path': '/',
+      ':method': 'GET',
+      ':scheme': 'http',
+      ':authority': 'localhost'
```

Comment:
`:authority` should include the port also... so `':authority': `localhost:${server.address().port}` here